apache/superset
# refactor(Dashboard): Fetch dashboard screenshot via dedicated endpoint

### SUMMARY
A dashboard can have an image attached in the markdown component that is hosted on a server that does not have proper CORS settings. Due to limitations of html2canvas, both PDF and image download will fail.

For reference from html2canvas:
> All the images that the script uses need to reside under the [same origin](http://en.wikipedia.org/wiki/Same_origin_policy) for it to be able to read them without the assistance of a [proxy](https://html2canvas.hertzen.com/proxy/). Similarly, if you have other canvas elements on the page, which have been tainted with cross-origin content, they will become dirty and no longer readable by html2canvas. 

This PR introduces the necessary frontend changes to defer the generation of the screenshot to a newly built endpoint. See https://github.com/apache/superset/pull/29187

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


### TESTING INSTRUCTIONS
1. Select a Dashboard and a specific tab
2. Download the dashboard as an image or PDF via the dedicated menu items
3. Wait for the download to complete
4. The screenshot should be of the chosen type and depict the target tab

### ADDITIONAL INFORMATION
